src/transformers/models/clvp/modeling_clvp.py

```diff
@@ -0,0 +1,1445 @@
+
+@add_start_docstrings(CLVP_START_DOCSTRING)
+class CLVPModel(CLVPPreTrainedModel):
+    config_class = CLVPConfig
+
+    def __init__(self, config: CLVPConfig):
+        super().__init__(config)
+
+        if not isinstance(config.text_config, CLVPTextConfig):
+            raise ValueError(
+                "config.text_config is expected to be of type CLVPTextConfig but is of type"
+                f" {type(config.text_config)}."
+            )
+
+        if not isinstance(config.speech_config, CLVPSpeechConfig):
+            raise ValueError(
+                "config.speech_config is expected to be of type CLVPSpeechConfig but is of type"
+                f" {type(config.speech_config)}."
+            )
+
+        text_config = config.text_config
+        speech_config = config.speech_config
+
+        self.projection_dim = config.projection_dim
+        self.text_embed_dim = text_config.hidden_size
+        self.speech_embed_dim = speech_config.hidden_size
+
+        self.text_model = CLVPTextTransformer(text_config)
+        self.speech_model = CLVPSpeechTransformer(speech_config)
+
+        self.text_projection = nn.Linear(self.text_embed_dim, self.projection_dim, bias=False)
+        self.speech_projection = nn.Linear(self.speech_embed_dim, self.projection_dim, bias=False)
+        self.logit_scale = nn.Parameter(torch.tensor(self.config.logit_scale_init_value))
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(CLVP_TEXT_INPUTS_DOCSTRING)
+    def get_text_features(
+        self,
+        input_ids: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        use_causal_attention_mask: Optional[bool] = False,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> torch.FloatTensor:
+        # Use CLVP model's config for some fields (if specified) instead of those of speech & text components.
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        text_outputs = self.text_model(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            use_causal_attention_mask=use_causal_attention_mask,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        pooled_output = text_outputs[1]
+        text_features = self.text_projection(pooled_output)
+
+        return text_features
+
+    @add_start_docstrings_to_model_forward(CLVP_SPEECH_INPUTS_DOCSTRING)
+    def get_speech_features(
```

Comment:
Missing code docstring for this method

Reply:
Just added the examples (below this block).